## Ćwiczenie 1: Series — tworzenie i operacje 

In [2]:
import pandas as pd
import numpy as np
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


# 1a

In [3]:
dane_pensja={'Anna': 5500,
'Jan': 7200,
'Ewa': 4800,
'Marek': 9100,
'Kasia': 6300
}
pensja = pd.Series(dane_pensja)
print(pensja)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


# 1b

In [4]:
najwiecej_zarabia = pensja.idxmax()
print(f'Najwięcej zarabia: {najwiecej_zarabia}.')
srednia = pensja.mean()
print(f'Średnia pensja wynosi: {srednia}.')

pow_sr = pensja[pensja>srednia]
print(f'Powyej średniej zarabiają: \n{pow_sr}')

ponizej_sr = pensja[pensja<6000]
klucze = len(ponizej_sr.keys())
print(f'{klucze} osób zarabia ponizej 6000zł')



Najwięcej zarabia: Marek.
Średnia pensja wynosi: 6580.0.
Powyej średniej zarabiają: 
Jan      7200
Marek    9100
dtype: int64
2 osób zarabia ponizej 6000zł


# 1c

In [5]:
po_podwyzce = (pensja * 1,10)
print(f'Pensje po podwyce o 10% wynoszą: \n{po_podwyzce}')

premia = pensja + 500 
print(f'Pensje po premii 500zł: \n{premia}')

nowa_pensja = (pensja * 1.10) + 500
print(f'Nowa pensja wynosi: \n{nowa_pensja}')

posortowane = nowa_pensja.sort_values(ascending=False)
print(f'Ranking nowych pensji od najwyszej: \n{posortowane}')



Pensje po podwyce o 10% wynoszą: 
(Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64, 10)
Pensje po premii 500zł: 
Anna     6000
Jan      7700
Ewa      5300
Marek    9600
Kasia    6800
dtype: int64
Nowa pensja wynosi: 
Anna      6550.0
Jan       8420.0
Ewa       5780.0
Marek    10510.0
Kasia     7430.0
dtype: float64
Ranking nowych pensji od najwyszej: 
Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


# Ćw.2

In [6]:
sklep = pd.DataFrame({
'produkt': ['Laptop','Tablet','Smartfon','Słuchawki','Monitor'],
'cena': [3500, 1800, 2500, 350, 2200],
'stan_magazynowy': [45, 120, 200, 500, 75],
'kategoria': ['Komputery','Mobilne','Mobilne','Akcesoria',
'Komputery']
})

# 2a

In [12]:
print(f'Kształt tabeli: {sklep.shape}')
print(f'Nazwy kolumn: {sklep.columns}')
print(f'Typ danych w kadej kolumnie \n {sklep.dtypes}')
print(f'Trzy pierwsze wiersze: \n{sklep.head(3)}')

Kształt tabeli: (5, 4)
Nazwy kolumn: Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')
Typ danych w kadej kolumnie 
 produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object
Trzy pierwsze wiersze: 
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


# 2b

In [ ]:
sklep['wartosc_magazynu']= sklep['cena'] * sklep['stan_magazynowy']
sklep['cena_z_vat']= sklep['cena'] * 1.23
print(sklep)

     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0
     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


In [20]:
url ='https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

# 2c

In [ ]:
print(f'\n Kształt: {tips.shape}')
print(f'\n Ostatnie pięć wierszy: {tips.tail(5)}')
print(f'\n Trzy losowe wiersze: {tips.sample(3)}')
print(f'\n Typ danych w kadej kolumnie: {tips.dtypes}')



 Kształt: (244, 7)

 Ostatnie pięć wierszy:      total_bill   tip     sex smoker   day    time  size
239       29.03  5.92    Male     No   Sat  Dinner     3
240       27.18  2.00  Female    Yes   Sat  Dinner     2
241       22.67  2.00    Male    Yes   Sat  Dinner     2
242       17.82  1.75    Male     No   Sat  Dinner     2
243       18.78  3.00  Female     No  Thur  Dinner     2

 Trzy losowe wiersze:      total_bill   tip     sex smoker  day    time  size
214       28.17  6.50  Female    Yes  Sat  Dinner     3
216       28.15  3.00    Male    Yes  Sat  Dinner     5
68        20.23  2.01    Male     No  Sat  Dinner     2

 Typ danych w kadej kolumnie: total_bill    float64
tip           float64
sex               str
smoker            str
day               str
time              str
size            int64
dtype: object


# Ćw.3 

In [26]:
url ='https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)

# 3a

In [34]:
print(penguins.head(10))
print(f'\n Kształt: {penguins.shape}')
print(penguins.info())

  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0    MALE  
6       36

# 3b

In [35]:
print(penguins.describe())

       bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      342.000000     342.000000         342.000000   342.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.459584       1.974793          14.061714   801.954536
min         32.100000      13.100000         172.000000  2700.000000
25%         39.225000      15.600000         190.000000  3550.000000
50%         44.450000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000


Średnia dlugość dzioba wynosi ok. 44 mm. Odchylenie standardowe równe ok. 5.45 mm mówi o umiarkowanie zrónicowanej długości dziobów. 


In [36]:
print(penguins.describe(include='all'))

       species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count      344     344      342.000000     342.000000         342.000000   
unique       3       3             NaN            NaN                NaN   
top     Adelie  Biscoe             NaN            NaN                NaN   
freq       152     168             NaN            NaN                NaN   
mean       NaN     NaN       43.921930      17.151170         200.915205   
std        NaN     NaN        5.459584       1.974793          14.061714   
min        NaN     NaN       32.100000      13.100000         172.000000   
25%        NaN     NaN       39.225000      15.600000         190.000000   
50%        NaN     NaN       44.450000      17.300000         197.000000   
75%        NaN     NaN       48.500000      18.700000         213.000000   
max        NaN     NaN       59.600000      21.500000         231.000000   

        body_mass_g   sex  
count    342.000000   333  
unique          NaN     2  
top

Dla kolumn tekstowych dodaje statystyki, których nie da się wyliczyc matematycznie: uniqe, top, freq. Pozwala to szybko wyróznić dominujące grupy.

In [39]:
mediana_masy= penguins['body_mass_g'].median()
print(f'Mediana masy ciała wynosi: {mediana_masy} g')

Mediana masy ciała wynosi: 4050.0 g


# 3c

In [45]:
podzial= penguins['species'].value_counts()
print(f'Ile pingwinów kadego gatunku?: \n{podzial}')

wyspa= penguins['island'].value_counts()
print(f'\n Ile pingwinów na kadej wyspie?: \n{wyspa}')

rozklad_plci= penguins['sex'].value_counts() 
print(f'\nRozkład płci: \n{rozklad_plci}')

Ile pingwinów kadego gatunku?: 
species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

 Ile pingwinów na kadej wyspie?: 
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

Rozkład płci: 
sex
MALE      168
FEMALE    165
Name: count, dtype: int64


Rozkad płci jest zbalansowany: samce - 168, samice - 165.

In [46]:
wyniki= penguins['species'].value_counts(normalize=True)*100
print(f"Gatunek Adelie stanowi {wyniki['Adelie']} %")

Gatunek Adelie stanowi 44.18604651162791 %


# 3d

In [49]:
braki= penguins.isna().sum()
print(f'Liczba brakujących wartości w kolumnach: \n{braki}')

braki_sex= (braki['sex']/len(penguins))*100
print(f'\n W kolumnie "sex" brakuje {braki_sex} % danych.')

Liczba brakujących wartości w kolumnach: 
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

 W kolumnie "sex" brakuje 3.1976744186046515 % danych.


Dataset zawiera 344 pingwinów z 3 gatunków.

Najczęstszy gatunek to: Adelie, 152 osobniki, 44.2%.

Liczba brakujących wartości: 19.

Średnia masa ciała wynosi: 4201.754386 g.

Pingwiny mieszkają na 3 wyspach, najliczniej na Biscoe.


# Ćw.4

In [50]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')
# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

# 4a

In [70]:
sr_rachunek= tips['total_bill'].mean()
print(f'Średni rachunek wynosi: {sr_rachunek:.2f}')
sr_napiwek_pr= tips['tip_pct'].mean()
print(f'\n Średni napiwek wynosi: {sr_napiwek_pr:.2f}%')
dzien= tips['day'].value_counts().idxmax()
print(f'\nW {dzien} występuje najwięcej rachunków')
palacze=tips['smoker'].value_counts()['Yes']
print(f'\n{palacze} rachunków jest od palaczy.')
mediana_napiwkow= tips['tip'].median()
print(f'\nMediana napiwku wynosi: {mediana_napiwkow}.')
podzial_czasu= tips['time'].value_counts(normalize=True)*100
print(f'\nProcent rachunków z obiadu: {podzial_czasu['Dinner']:.2f}%.')
rozmiary=tips['size'].nunique()
print(f'\nW danych występuje {rozmiary} unikalnych rozmiarów grup.')




Średni rachunek wynosi: 19.79

 Średni napiwek wynosi: 16.08%

W Sat występuje najwięcej rachunków

93 rachunków jest od palaczy.

Mediana napiwku wynosi: 2.9.

Procent rachunków z obiadu: 72.13%.

W danych występuje 6 unikalnych rozmiarów grup.
